cmd/src/campaigns_new.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"text/template"
+
+	"github.com/pkg/errors"
+	"github.com/sourcegraph/src-cli/internal/campaigns"
+)
+
+func init() {
+	usage := `
+'src campaigns new' creates a new campaign spec YAML, prefilled with all
+required fields.
+
+Usage:
+
+    src campaigns new [-f FILE]
+
+Examples:
+
+
+    $ src campaigns new -f campaign.spec.yaml
+
+`
+
+	flagSet := flag.NewFlagSet("new", flag.ExitOnError)
+
+	var (
+		fileFlag = flagSet.String("f", "campaign.yaml", "The name of campaign spec file to create.")
+	)
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		if _, err := os.Stat(*fileFlag); !os.IsNotExist(err) {
+			return fmt.Errorf("file %s already exists", *fileFlag)
+		}
+
+		f, err := os.Create(*fileFlag)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create file %s", *fileFlag)
+		}
+		defer f.Close()
+
+		tmpl, err := template.New("").Parse(campaignSpecTmpl)
+		if err != nil {
+			return err
+		}
+
+		author := campaigns.GitCommitAuthor{
+			Name:  "Sourcegraph",
+			Email: "campaigns@sourcegraph.com",
+		}
+
+		// Try to get better default values from git, ignore any errors.
+		if err := checkExecutable("git", "version"); err == nil {
```

Comment:
is this to test if git is available?

Reply:
Yeah. Strictly speaking it's not necessary since we ignore the other errors anyway, but I like this check more.